src/app/inventory/d2-stores.ts

```diff
@@ -141,12 +150,111 @@ export function loadStores(): ThunkResult<DimStore[] | undefined> {
     }
 
     $featureFlags.clarityDescriptions && dispatch(loadClarity()); // no need to await
+    await dispatch(loadNewItems(account));
     const stores = await dispatch(loadStoresData(account));
     return stores;
   };
 }
 
-let latestDateLastPlayedTimestamp = 0;
+const BUNGIE_CACHE_TTL = 15_000;
+
+let minimumCacheAge = Number.MAX_SAFE_INTEGER;
+
+function loadProfile(account: DestinyAccount): ThunkResult<DestinyProfileResponse | undefined> {
+  return async (dispatch, getState) => {
+    const mockProfileData = getState().inventory.mockProfileData;
+    if (mockProfileData) {
+      // TODO: can/should we replace this with profileResponse plus the readOnly flag?
+      return mockProfileData;
+    }
+
+    // First try loading from IndexedDB
+    let profileResponse = getState().inventory.profileResponse;
+    if (!profileResponse) {
+      profileResponse = await get<DestinyProfileResponse>(`profile-${account.membershipId}`);
+      // Check to make sure the profile hadn't been loaded in the meantime
+      if (getState().inventory.profileResponse) {
+        profileResponse = getState().inventory.profileResponse;
+      } else {
+        infoLog('d2-stores', 'Loaded cached profile from IndexedDB');
+        dispatch(profileLoaded({ profile: profileResponse, live: false }));
+      }
+    }
+
+    let cachedProfileMintedDate = new Date(0);
+
+    // If our cached profile is up to date
+    if (profileResponse) {
+      // TODO: need to make sure we still load at the right frequency / for manual cache busts?
+      cachedProfileMintedDate = new Date(profileResponse.responseMintedTimestamp ?? 0);
+      const profileAge = Date.now() - cachedProfileMintedDate.getTime();
+      if (!storesLoadedSelector(getState()) && profileAge > 0 && profileAge < BUNGIE_CACHE_TTL) {
+        warnLog(
+          'd2-stores',
+          'Cached profile is within Bungie.net cache time, skipping remote load.',
+          profileAge
+        );
+        return profileResponse;
+      } else {
+        warnLog(
+          'd2-stores',
+          'Cached profile is older than Bungie.net cache time, proceeding.',
+          profileAge
+        );
+      }
+    }
+
+    try {
+      const remoteProfileResponse = await getStores(account);
+      const remoteProfileMintedDate = new Date(remoteProfileResponse.responseMintedTimestamp ?? 0);
+
+      // compare new response against cached response, toss if it's not newer!
+      if (
```

Comment:
personally wrapping 212 to 237 in an `if (profileResponse)` would make it an iota more readable